nifi-nar-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/SnowflakeComputingConnectionPool.java

```diff
@@ -0,0 +1,314 @@
+        dataSource.setMinEvictableIdleTimeMillis(minEvictableIdleTimeMillis);
+        dataSource.setSoftMinEvictableIdleTimeMillis(softMinEvictableIdleTimeMillis);
+
+        if (validationQuery != null && !validationQuery.isEmpty()) {
+            dataSource.setValidationQuery(validationQuery);
+            dataSource.setTestOnBorrow(true);
+        }
+
+        final List<PropertyDescriptor> dynamicProperties = context.getProperties()
+            .keySet()
+            .stream()
+            .filter(PropertyDescriptor::isDynamic)
+            .collect(Collectors.toList());
+
+        dynamicProperties.forEach((descriptor) -> {
+            final PropertyValue propertyValue = context.getProperty(descriptor);
+            if (descriptor.isSensitive()) {
+                final String propertyName = StringUtils.substringAfter(descriptor.getName(), SENSITIVE_PROPERTY_PREFIX);
+                dataSource.addConnectionProperty(propertyName, propertyValue.getValue());
+            } else {
+                dataSource.addConnectionProperty(descriptor.getName(), propertyValue.evaluateAttributeExpressions().getValue());
+            }
+        });
+    }
+
+    protected Driver getDriver(final String driverName, final String url) {
+        final Class<?> clazz;
+
+        try {
+            clazz = Class.forName(driverName);
+        } catch (final ClassNotFoundException e) {
+            throw new ProcessException("Driver class " + driverName +  " is not found", e);
+        }
+
+        try {
+            return DriverManager.getDriver(url);
+        } catch (final SQLException e) {
+            // In case the driver is not registered by the implementation, we explicitly try to register it.
+            try {
+                final Driver driver = (Driver) clazz.newInstance();
+                DriverManager.registerDriver(driver);
+                return DriverManager.getDriver(url);
+            } catch (final SQLException e2) {
+                throw new ProcessException("No suitable driver for the given Database Connection URL", e2);
+            } catch (final IllegalAccessException | InstantiationException e2) {
+                throw new ProcessException("Creating driver instance is failed", e2);
+            }
+        }
+    }
+
+    /**
+     * Shutdown pool, close all open connections.
+     *
+     * @throws SQLException if there is an error while closing open connections
+     */
+    @OnDisabled
+    public void shutdown() throws SQLException {
+        try {
+            if (dataSource != null) {
+                dataSource.close();
+            }
+        } finally {
+            dataSource = null;
+        }
+    }
+
+    @Override
+    public Connection getConnection() throws ProcessException {
+        try {
+            final Connection connection = dataSource.getConnection();
+            return connection;
+        } catch (final SQLException e) {
+            throw new ProcessException(e);
+        }
+    }
```

Comment:
The `shutdown` and `getConnection` methods in `DBCPConnectionPool` should work since they perform conditional checks on the Kerberos user property and otherwise implement the same behavior.

Reply:
Yeah, sounds reasonable.